apps/sms/js/threads.js

```diff
@@ -14,12 +14,12 @@
       this[key] = thread[key];
     }
 
-    this.messages = [];
+    this.messages = new Map();
```

Comment:
my main problem with using a map is that we lose the ordering we have with arrays... While we don't care about ordering in JS right now, I think we really need it in the future: ordering needs to be done in JS rather than using the DOM.

So maybe we need both an array and a Map. That said I'm fine if we look at this issue later.